src/reanimated2/hook/useAnimatedStyle.ts

```diff
@@ -513,9 +513,9 @@ For more, see the docs: \`https://docs.swmansion.com/react-native-reanimated/doc
 
   checkSharedValueUsage(initial.value);
 
-  if (isJest()) {
-    return { viewDescriptors, initial, viewsRef, animatedStyle };
-  } else {
-    return { viewDescriptors, initial, viewsRef };
-  }
+  const style = isJest()
+    ? { viewDescriptors, initial, viewsRef, animatedStyle }
+    : { initial, viewsRef, viewDescriptors };
+
+  return useRef(style).current;
```

Comment:
I'm a bit concerned about these changes and want to make sure they don't mess up rendering, hot reload, or fast refresh. Before we merge them, we need to figure out the best way to test everything. What do you suggest for testing? I want to avoid any unexpected issues.

Reply:
We should run our tests on all the platforms:
- IOS (paper & fabric)
- Android (paper & fabric)
- Web
- TVOS ?
- MacOS ?

And what to test? My suggestions are:
- Simple animated style (number, string & color)
- Inline animated style (number, string & color)
- Animated style inside navigation
- Refresh and reload

Reply:
Seems to be ok, but let's wait until we merge https://github.com/software-mansion/react-native-reanimated/pull/5458

Reply:
Hey @piaskowyk is this ready to merge now? Or it's going to be changed as the same way as `useSharedValue` implementation being change to `useState` instead of `useRef`?

Reply:
@ranaavneet I don't think there's a reason to use `useState` here instead of `useRef` - we don't have any heavy computation in `useRef` argument here.

Reply:
Thanks @tjzel! Any idea, when this improvement would be available ?

Reply:
I see no reason why shouldn't we include it in 3.7.0 - but when will that be? Definitely next year 😅

Reply:
While it's only a few lines of code I don't feel like shipping such significant change in a patch would be a good idea.

Reply:
Makes sense! I guess a lot of PR's that i am interested in, would land in 3.7.0 Like #5333 #4300 #5505 #5462 #5458

Reply:
Why no hype for #5500 😭

Reply:
@ranaavneet @tjzel is testing done for this? Can I put these changes as patches in my project?

Reply:
Not yet, most of the team was OOO in the Christmas-New year period.

Reply:
Hey @tjzel any update on this?